front/src/routes/integration/all/withings/settings-page/actions.js

```diff
@@ -0,0 +1,55 @@
+import queryString from 'query-string';
+import { route } from 'preact-router';
+import { RequestStatus } from '../../../../../utils/consts';
+
+const actions = store => ({
+  async initWithingsDevices(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting,
+      withingsGetStatus: RequestStatus.Getting
+    });
+    try {
+      // check if this call is a return of oauth2 authorize code
+      if (state.currentUrl) {
+        const queryParams = queryString.parse(state.currentUrl.substring(state.currentUrl.indexOf('?')));
```

Comment:
I'm not sure we need that dependency, I think preact is already parsing the GET parameters.

Reply:
i fix it